src/Mvc/Mvc.RazorPages/src/DependencyInjection/PageConventionCollectionExtensions.cs

```diff
@@ -313,13 +371,23 @@ public static PageConventionCollection AuthorizeFolder(this PageConventionCollec
                 throw new ArgumentException(Resources.ArgumentCannotBeNullOrEmpty, nameof(folderPath));
             }
 
-            var authorizeFilter = new AuthorizeFilter(policy);
-            conventions.AddFolderApplicationModelConvention(folderPath, model => model.Filters.Add(authorizeFilter));
+            var authorizeAttribute = new AuthorizeAttribute(policy);
```

Comment:
is this attribute instance used anywhere?  the endpoint metadata is newing another one up below